assets/scripts/dialogs/GeotagDialog.jsx

```diff
@@ -48,148 +44,138 @@ L.Icon.Default.mergeOptions({
   shadowUrl: '/images/marker-shadow.png'
 })
 
-/**
-This Dialog uses the generic Dialog component and combines it with the GeoSearch
-and LocationPopup components as well as a map to display the coordinates on
-It handles setting, displaying, and clearing location information assocaited with a 'street'
- */
-
-class GeotagDialog extends React.Component {
-  static propTypes = {
-    // Provided by react-intl higher-order component
-    intl: PropTypes.object.isRequired,
-
-    // Provided by Redux store
-    street: PropTypes.object,
-    markerLocation: PropTypes.shape({
-      lat: PropTypes.number,
-      lng: PropTypes.number
-    }),
-    addressInformation: PropTypes.object,
-    userLocation: PropTypes.shape({
-      latitude: PropTypes.number,
-      longitude: PropTypes.number
-    }),
-    dpi: PropTypes.number,
+GeotagDialog.propTypes = {
+  // Provided by Redux store
+  street: PropTypes.object,
+  markerLocation: PropTypes.shape({
+    lat: PropTypes.number,
+    lng: PropTypes.number
+  }),
+  addressInformation: PropTypes.object,
+  userLocation: PropTypes.shape({
+    latitude: PropTypes.number,
+    longitude: PropTypes.number
+  })
+}
 
-    // Provided by Redux action dispatchers
-    setMapState: PropTypes.func,
-    addLocation: PropTypes.func,
-    clearLocation: PropTypes.func,
-    saveStreetName: PropTypes.func
+function getInitialState (props) {
```

Comment:
This function still bothers me, but I think it might be fine/isn't negatively affecting performance or anything. The only incremental improvement I could think was ```mapCenter = markerLocation = props.street.location.latlng``` for the first two if statements, but that sort of consolidation feels just as likely to annoy future readers. :-D